src/app/applications/multicast/multicast-detail/multicast-detail.component.ts

```diff
@@ -0,0 +1,203 @@
+  private deleteDialogSubscription: Subscription;
+  public dropdownButton: DropdownButton;
+
+  public formFailedSubmit: boolean = false;
+  private applicationId: number;
+  public downlink = new Downlink();
+  @Input() errorMessages: string[];
+
+  constructor(
+    private route: ActivatedRoute,
+    private snackBar: MatSnackBar,
+    public dialog: MatDialog,
+    private deleteDialogService: DeleteDialogService,
+    private location: Location,
+    private multicastService: MulticastService,
+    public translate: TranslateService,
+    public snackService: SnackService,
+    public downlinkService: DownlinkService,
+    private errorMessageService: ErrorMessageService
+  ) {}
+
+  ngOnInit(): void {
+    this.errorMessages = [];
+    const id: number = +this.route.snapshot.paramMap.get('multicastId');
+    if (id) {
+      this.getMulticast(id);
+      this.dropdownButton = {
+        label: '',
+        editRouterLink: '../../multicast-edit/' + id,
+        isErasable: true,
+      };
+      this.applicationId = +this.route.snapshot.paramMap.get('id');
+    }
+    this.translate
+      .get(['GEN.BACK', 'MULTICAST-TABLE-ROW.SHOW-OPTIONS'])
+      .subscribe((translations) => {
+        this.backButton.label = translations['GEN.BACK'];
+        this.dropdownButton.label =
+          translations['MULTICAST-TABLE-ROW.SHOW-OPTIONS'];
+      });
+  }
+
+  getMulticast(id: number) {
+    this.multicastService.get(id).subscribe((multicast: Multicast) => {
+      this.multicast = multicast;
+      this.setBackButton(this.applicationId);
+    });
+  }
+
+  private setBackButton(applicationId: number) {
+    this.backButton.routerLink = ['applications', applicationId.toString()];
+  }
+
+  //only if classB can be used
+  // canShowPeriodicity(): boolean {
+  //   if (this.multicast.groupType === MulticastType.ClassB) {
+  //     return true;
+  //   } else return false;
+  // }
+
+  onDeleteMulticast() {
+    this.deleteDialogSubscription = this.deleteDialogService
+      .showSimpleDialog()
+      .subscribe((response) => {
+        if (response) {
+          this.multicastService
+            .delete(this.multicast.id)
+            .subscribe((response) => {
+              if (response.status !== 0) {
+                this.showDeletedSnack();
+                this.location.back();
+              } else {
+                this.showFailSnack();
+              }
+            });
+        } else {
+          console.log(response);
+        }
+      });
+  }
+
+  showDeletedSnack(): void {
+    this.snackService.showDeletedSnack();
+  }
+
+  showFailSnack(): void {
+    this.snackService.showFailSnack();
+  }
+  showQueueSnack(): void {
+    this.snackService.showInQueueSnack();
+  }
+  keyPressHexadecimal(event) {
+    // make sure only hexadecimal can be typed in input with adresses.
+    var inp = String.fromCharCode(event.keyCode);
+
+    if (/[a-fA-F0-9]/.test(inp)) {
```

Comment:
Refactor the regular expression. it's used both here and in the *.edit.component